tests/hikari/impl/test_rest.py

```diff
@@ -1031,12 +1032,19 @@ def test_unban_member(self, rest_client: rest.RESTClientImpl):
         mock_unban_user.assert_called_once_with(123, 321, reason="ayaya")
 
     def test_command_builder(self, rest_client):
-        result = rest_client.command_builder("a name", "very very good")
-
-        assert isinstance(result, special_endpoints.CommandBuilder)
+        result = rest_client.command_builder(1, "a name", description="very very good")
+        assert isinstance(result, special_endpoints.SlashCommandBuilder)
         assert result.name == "a name"
         assert result.description == "very very good"
 
+        result = rest_client.command_builder(2, "a name")
+        assert isinstance(result, special_endpoints.ContextMenuCommandBuilder)
+        assert result.name == "a name"
+        assert result.description is undefined.UNDEFINED
+
+        with pytest.raises(TypeError):
+            rest_client.command_builder(1, "a name")
```

Comment:
These are 3 test cases in 1, it should be split

Also, the tests should make use of the enum and not hard-code the numbers if they are relevant to the test case